### PR TITLE
task done;

### PR DESCRIPTION
**Pull request by AndrewOliferchuk**

### CODE REVIEW

File: src/main/java/com/example/demo/config/SecurityConfig.java [hunk 1]
```diff
@@ -2,22 +2,30 @@
...
 
 @EnableMethodSecurity
 @Configuration
 public class SecurityConfig {
-    private UserDetailsService userDetailsService;
+    private final JwtAuthenticationFilter jwtAuthenticationFilter;
+    private final UserDetailsService userDetailsService;
 
-    public SecurityConfig(UserDetailsService userDetailsService) {
+    public SecurityConfig(JwtAuthenticationFilter jwtAuthenticationFilter,
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/config/SecurityConfig.java:26]: Let's use Lombok annotation `@RequiredArgsConstructor`

File: src/main/java/com/example/demo/controller/AuthenticationController.java [hunk 2]
```diff
@@ -16,6 +19,12 @@
 @RequestMapping(value = "/auth")
 public class AuthenticationController {
     private UserService userService;
+    private AuthenticationService authenticationService;
+
+    @PostMapping("/login")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/controller/AuthenticationController.java]:
```suggestion
    public UserLoginResponseDto login(@RequestBody @Valid UserLoginRequestDto requestDto) {
```

File: src/main/java/com/example/demo/controller/AuthenticationController.java [hunk 3]
```diff
@@ -16,6 +19,12 @@
 @RequestMapping(value = "/auth")
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/controller/AuthenticationController.java]:
```suggestion
@RequestMapping("/auth")
```

File: src/main/java/com/example/demo/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            Authentication authentication = new UsernamePasswordAuthenticationToken(userDetails,
+                    null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    public String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/security/JwtAuthenticationFilter.java]: Let's use Spring constant HttpHeaders.AUTHORIZATION

File: src/main/java/com/example/demo/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/demo/security/JwtAuthenticationFilter.java]: Create constant for "Bearer "

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

File: src/main/java/com/example/demo/config/SecurityConfig.java (hunk 1 above)
Comment (AndrewOliferchuk) [reply src/main/java/com/example/demo/config/SecurityConfig.java:26]: The code does not work with the annotation, but it works with the constructor

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**